src/ember-app.js

```diff
@@ -287,77 +111,203 @@ class EmberApp {
+  *
+  * @param {string} distPath path to the built Ember app to load
+  * @param {Sandbox} [sandboxClass=VMSandbox] sandbox class to use
+  * @param {Object} [sandboxGlobals={}] any additional variables to expose in the sandbox or override existing in the sandbox
+  */
+function buildSandbox(distPath, sandboxClass, sandboxGlobals, moduleWhitelist, appConfig) {
+  let Sandbox = sandboxClass || require('./vm-sandbox');
+  let sandboxRequire = buildWhitelistedRequire(moduleWhitelist, distPath);
+  let config = appConfig;
+
+  // add any additional user provided variables or override the default globals in the sandbox
+  let globals = {
+    najax: najax,
+    FastBoot: {
+      require: sandboxRequire,
+      config: () => ({ default: config })
+    }
+  };
+
+  for (let key in sandboxGlobals) {
+    if (sandboxGlobals.hasOwnProperty(key)) {
+      globals[key] = sandboxGlobals[key];
     }
+  }
 
-    let manifest;
-    let pkg;
+  return new Sandbox({ globals });
+}
 
-    try {
-      pkg = JSON.parse(file);
-      manifest = pkg.fastboot.manifest;
-    } catch (e) {
-      throw new Error(`${pkgPath} was malformed or did not contain a manifest. Ensure that you have a compatible version of ember-cli-fastboot.`);
+/**
+  * The Ember app runs inside a sandbox that doesn't have access to the normal
+  * Node.js environment, including the `require` function. Instead, we provide
+  * our own `require` method that only allows whitelisted packages to be
+  * requested.
+  *
+  * This method takes an array of whitelisted package names and the path to the
+  * built Ember app and constructs this "fake" `require` function that gets made
+  * available globally inside the sandbox.
+  *
+  * @param {string[]} whitelist array of whitelisted package names
+  * @param {string} distPath path to the built Ember app
+  */
+function buildWhitelistedRequire(whitelist, distPath) {
+  whitelist.forEach(function(whitelistedModule) {
+    debug("module whitelisted; module=%s", whitelistedModule);
+  });
+
+  return function(moduleName) {
+    if (whitelist.indexOf(moduleName) > -1) {
+      let nodeModulesPath = path.join(distPath, 'node_modules', moduleName);
+
+      if (existsSync(nodeModulesPath)) {
+        return require(nodeModulesPath);
+      } else {
+        // If it's not on disk, assume it's a built-in node package
+        return require(moduleName);
+      }
+    } else {
+      throw new Error("Unable to require module '" + moduleName + "' because it was not in the whitelist.");
     }
+  };
+}
 
-    return {
-      appFile:  path.join(distPath, manifest.appFile),
-      vendorFile: path.join(distPath, manifest.vendorFile),
-      htmlFile: path.join(distPath, manifest.htmlFile),
-      moduleWhitelist: pkg.fastboot.moduleWhitelist,
-      hostWhitelist: pkg.fastboot.hostWhitelist,
-      appConfig: pkg.fastboot.appConfig
-    };
+/**
+  * Given the path to a built Ember app, reads the FastBoot manifest
+  * information from its `package.json` file.
```

Comment:
I can add JS docs, but since this is not on the class itself, it's not public or private API